osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -289,18 +290,24 @@ private void beatmapSetsChanged(IRealmCollection<BeatmapSetInfo> sender, ChangeS
             {
                 foreach (int i in changes.DeletedIndices.OrderDescending())
                 {
-                    setsRequiringRemoval.Add(realmBeatmapSets[i]);
+                    Guid id = realmBeatmapSets[i];
+
+                    setsRequiringRemoval.Add(id);
+                    setsRequiringUpdate.Remove(id);
```

Comment:
My primary question here, lacking most of context et cetera, would be - why do deletions need to care about pending updates, but updates don't need to care about pending deletions?

Reply:
You're right. It can/should likely be handled in both directions. Just a lot harder to trigger that case.

Reply:
I'm not sure if the above is a "good point, will get fixed" comment or a "good point, will not get fixed until demonstrated as a problem" comment. So I don't know if this is ready to go or not.

Reply:
I've made this change, but can't find an immediate way to test it. Should be fine I think.